relay.go

```diff
@@ -50,23 +52,26 @@ func (r *Relayer) Hosts() RelayHosts {
 
 // Relay forwards a frame.
 func (r *Relayer) Relay(f *Frame) error {
-	// TODO: remove relay items on timeout.
 	if f.messageType() != messageTypeCallReq {
-		r.RLock()
-		item, ok := r.items[f.Header.ID]
-		r.RUnlock()
-		if !ok {
-			return errors.New("non-callReq for inactive ID")
-		}
-		f.Header.ID = item.remapID
-		item.destination.Receive(f)
-		if f.isLast() {
-			r.removeRelayItem(f.Header.ID)
-		}
-		return nil
+		return r.handleNonCallReq(f)
+	}
+	return r.handleCallReq(f)
+}
+
+// Receive accepts a relayed frame.
+func (r *Relayer) Receive(f *Frame) {
+	r.RLock()
+	_, ok := r.items[f.Header.ID]
+	r.RUnlock()
+
+	r.conn.sendCh <- f
```

Comment:
For now, drop the frame (and log, since it shouldn't happen).

Ping req/res also need relay items (since they need ID remapping), let's do that as a separate change though.